bin/reth/src/node/mod.rs

```diff
@@ -335,6 +335,8 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
             transaction_pool.clone(),
             ctx.task_executor.clone(),
             Arc::clone(&self.chain),
+            #[cfg(feature = "optimism")]
```

Comment:
I think we also want a default ethereum feature and a compile_err

https://doc.rust-lang.org/std/macro.compile_error.html

if both features are enabled

Reply:
Ideally, the binary will still be able to run as an L1 client with some deliberation around where we use feature flags vs. the runtime checks - are we sure we want this? Down to keep the feature flag purpose-driven if that's what y'all would prefer, `op-geth` favors runtime feature flags (which we can do off of the chainspec's `optimism` field) in most places to support backwards compatibility with L1 though.

On the other hand, having a default `ethereum` feature flag would be nice for other chains that eventually want to integrate with reth, might save everyone some headache down the road :)

Reply:
Opened a PR to test this idea: https://github.com/anton-rs/op-reth/pull/74

Unfortunately, we can't support mutually exclusive flags as I understand since there isn't a way to do this without breaking any runs with `--all-features` (of which a lot of ci depends on)